threedi_modelchecker/checks/other.py

```diff
@@ -44,6 +44,141 @@ def description(self):
         )
 
 
+class CrossSectionSameConfigurationCheck(BaseCheck):
+    """Check the cross-sections on the object are either all open or all closed."""
+
+    def get_invalid(self, session):
+        # get all channels with more than 1 cross section location
+        cross_sections = (
+            select(
+                models.CrossSectionLocation.id.label("cross_section_id"),
+                models.CrossSectionLocation.channel_id,
+                models.CrossSectionDefinition.shape,
+                models.CrossSectionDefinition.width,
+                models.CrossSectionDefinition.height,
+                # dirty hack to get the first number in a space-separated list
+                cast(
+                    func.substr(
+                        models.CrossSectionDefinition.width,
+                        1,
+                        func.instr(models.CrossSectionDefinition.width, " ") - 1,
+                    ),
+                    REAL,
+                ).label("first_width"),
+                cast(
+                    func.substr(
+                        models.CrossSectionDefinition.height,
+                        1,
+                        func.instr(models.CrossSectionDefinition.height, " ") - 1,
+                    ),
+                    REAL,
+                ).label("first_height"),
+                # even dirtier hack to get the last number in a space-separated list
+                cast(
+                    func.replace(
+                        models.CrossSectionDefinition.width,
+                        func.rtrim(
+                            models.CrossSectionDefinition.width,
+                            func.replace(models.CrossSectionDefinition.width, " ", ""),
+                        ),
+                        "",
+                    ),
+                    REAL,
+                ).label("last_width"),
+                cast(
+                    func.replace(
+                        models.CrossSectionDefinition.height,
+                        func.rtrim(
+                            models.CrossSectionDefinition.height,
+                            func.replace(models.CrossSectionDefinition.height, " ", ""),
+                        ),
+                        "",
+                    ),
+                    REAL,
+                ).label("last_height"),
+            )
+            .select_from(models.CrossSectionLocation)
+            .join(
+                models.CrossSectionDefinition,
+                models.CrossSectionLocation.definition_id
+                == models.CrossSectionDefinition.id,
+            )
+            .subquery()
+        )
+        cross_sections_with_configuration = select(
+            cross_sections.c.cross_section_id,
+            cross_sections.c.shape,
+            cross_sections.c.last_width,
+            cross_sections.c.channel_id,
+            case(
```

Comment:
I think it is good to split this out in a function too. This would enable you to shorten the variables and avoid some line breaks. I guess you would need 5 arguments? (shape, first_width, last_width, first_height, last_height)